client/maps/actions.ts

```diff
@@ -0,0 +1,341 @@
+import {
+  GetMapDetailsPayload,
+  GetMapsPayload,
+  MapInfoJson,
+  MapPreferences,
+  MapSortType,
+  MapVisibility,
+  Tileset,
+  UpdateMapPayload,
+  UpdateMapServerBody,
+  UploadMapPayload,
+} from '../../common/maps'
+import { BaseFetchFailure } from '../network/fetch-action-types'
+
+export type MapsActions =
+  | UploadLocalMapBegin
+  | UploadLocalMapSuccess
+  | UploadLocalMapFailure
+  | GetMapsBegin
+  | GetMapsSuccess
+  | GetMapsFailure
+  | ToggleFavoriteMapBegin
+  | ToggleFavoriteMapSuccess
+  | ToggleFavoriteMapFailure
+  | RemoveMapBegin
+  | RemoveMapSuccess
+  | RemoveMapFailure
+  | RegenMapImageBegin
+  | RegenMapImageSuccess
+  | RegenMapImageFailure
+  | ClearMaps
+  | GetMapDetailsBegin
+  | GetMapDetailsSuccess
+  | GetMapDetailsFailure
+  | UpdateMapBegin
+  | UpdateMapSuccess
+  | UpdateMapFailure
+  | GetMapPreferencesBegin
+  | GetMapPreferencesSuccess
+  | GetMapPreferencesFailure
+  | UpdateMapPreferencesBegin
+  | UpdateMapPreferencesSuccess
+  | UpdateMapPreferencesFailure
+
+/**
+ * A request is being made to the server to upload a local map.
+ */
+export interface UploadLocalMapBegin {
+  type: '@maps/uploadLocalMapBegin'
+  payload: {
+    path: string
+  }
+}
+
+/**
+ * The server has returned the map that was uploaded.
+ */
+export interface UploadLocalMapSuccess {
+  type: '@maps/uploadLocalMap'
+  payload: UploadMapPayload
+  error?: false
+  meta: {
+    path: string
+  }
+}
+
+/**
+ * A request to upload a local map has failed.
+ */
+export interface UploadLocalMapFailure extends BaseFetchFailure<'@maps/uploadLocalMap'> {
+  meta: {
+    path: string
+  }
+}
+
+/**
+ * A request is being made to the server to retrieve the list of maps with a particular visibility
+ * and other filters.
+ */
+export interface GetMapsBegin {
+  type: '@maps/getMapsBegin'
+  payload: {
+    visibility: MapVisibility
+    limit: number
+    page: number
+    sort: MapSortType
+    numPlayers: number
+    tileset: Tileset
+    searchQuery: string
+  }
+}
+
+/**
+ * The server has returned the list of maps with a particular visibility and other filters applied.
+ */
+export interface GetMapsSuccess {
+  type: '@maps/getMaps'
+  payload: GetMapsPayload
+  error?: false
+  meta: {
+    visibility: MapVisibility
+    limit: number
+    page: number
+    sort: MapSortType
+    numPlayers: number
+    tileset: Tileset
+    searchQuery: string
+  }
+}
+
+/**
+ * A request to get the list of maps has failed.
+ */
+export interface GetMapsFailure extends BaseFetchFailure<'@maps/getMaps'> {
+  meta: {
+    visibility: MapVisibility
+    limit: number
+    page: number
+    sort: MapSortType
+    numPlayers: number
+    tileset: Tileset
+    searchQuery: string
+  }
+}
+
+/**
+ * A request is being made to the server to toggle the favorite status of a map.
+ */
+export interface ToggleFavoriteMapBegin {
+  type: '@maps/toggleFavoriteMapBegin'
```

Comment:
nit: we already know this is for `@maps`, can't we just shorten this to `toggleFavoriteBegin` and `toggleFavorite`?